core/src/main/java/org/apache/calcite/schema/lookup/Lookup.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.schema.lookup;
+
+import org.apache.calcite.util.NameMap;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import java.util.Set;
+import java.util.function.BiFunction;
+
+/**
+ * A casesensitive/insensitive lookup for tables, schemas, functions, types ...
+ *
+ * @param <T> Element type
+ */
+public interface Lookup<T> {
+  /**
+   * Returns a named entity with a given name, or null if not found.
+   *
+   * @param name Name
+   * @return Entity, or null
+   */
+  @Nullable T get(String name);
+
+  /**
+   * Returns a named entity with a given name ignoring the case, or null if not found.
+   *
+   * @param name Name
+   * @return Entity, or null
```

Comment:
particularly interface members can have longer documentation - e.g., as I wrote in an example above.

Reply:
OK